pkg/rate/api_limiter_norace_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !race,!privileged_tests
+
+package rate
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/cilium/cilium/pkg/testutils"
+	"golang.org/x/sync/semaphore"
+	"gopkg.in/check.v1"
+)
+
+// This test currently fails consistently when run under the race detector, see
+// https://github.com/cilium/cilium/issues/13508
+func (b *ControllerSuite) TestStressRateLimiter(c *check.C) {
+	a := NewAPILimiter("foo", APILimiterParameters{
+		EstimatedProcessingDuration: 5 * time.Millisecond,
+		RateLimit:                   1000.0,
+		ParallelRequests:            50,
+		RateBurst:                   1,
+		MaxWaitDuration:             10 * time.Millisecond,
+		AutoAdjust:                  true,
+	}, nil)
+
+	var (
+		sem                = semaphore.NewWeighted(100)
+		completed, retries int32
+	)
+
+	go func() {
+		for i := 0; i < 1000; i++ {
+			sem.Acquire(context.Background(), 1)
```

Comment:
The reason I think it fails is because the number of parallel go routines with we running. There was a similar issue with `AllocatedID`: https://github.com/cilium/cilium/blob/060901444c152cd293744a4af6a5b393299dd05d/pkg/idpool/idpool_race_test.go#L23-L30

https://github.com/cilium/cilium/blob/060901444c152cd293744a4af6a5b393299dd05d/pkg/idpool/idpool_norace_test.go#L23-L25

Reply:
Thanks for the hint. I'll try adjusting the test accordingly and see if it helps.